--- /tmp/opencode/gen-orig.cmake	2026-08-06 22:10:07.018934336 +0300
+++ /tmp/opencode/gen-new.cmake	2026-08-06 22:10:07.054198610 +0300
@@ -43,10 +43,8 @@
         set(subset_headers "")
         foreach (entry ${lang_sources})
             if (entry MATCHES "\\.(cpp|mm)$")
-                file(RELATIVE_PATH relative ${src_loc} ${entry})
-                list(APPEND subset_headers ${gen_dst}/lang_subsets/${relative}.h)
                 set_property(SOURCE ${entry} APPEND PROPERTY COMPILE_DEFINITIONS
-                    "LANG_KEYS_SUBSET=\"lang_subsets/${relative}.h\"")
+                    "LANG_KEYS_SUBSET=\"lang_auto_keys.h\"")
             endif()
         endforeach()
 
